### ISSUE TITLE
Old screen transitions are used on Android API 28

### ISSUE THREAD

**Issue by noelmansour** [flutter/flutter#21191]
Here's what a slowed down version of an activity transition looks like on Android API 27. It's a slide up + fade in animation:
![api27-transition](https://user-images.githubusercontent.com/327631/44821063-adef9e80-abc1-11e8-834c-4f4f38d9445e.gif)

After #13736 is fixed (or using `MaterialPageRoute.debugEnableFadingRoutes = true`), this should match Flutter's current behavior.

However, API 28 uses a different transition. It's a reveal + more subtle slide up animation:
![api28-transition](https://user-images.githubusercontent.com/327631/44821147-0aeb5480-abc2-11e8-9f89-260848b8ed74.gif)

`MaterialPageRoute` should use a transition that is consistent with the API level.

**Comment by zoechi** [comment]
Please add the output of `flutter doctor -v`.

**Comment by noelmansour** [comment]
Here's the output:

```
[✓] Flutter (Channel beta, v0.6.0, on Mac OS X 10.13.6 17G65, locale en-CA)
    • Flutter version 0.6.0 at /Users/noel/dev/flutter
    • Framework revision 9299c02cf7 (2 weeks ago), 2018-08-16 00:35:12 +0200
    • Engine revision e3687f70c7
    • Dart version 2.1.0-dev.0.0.flutter-be6309690f

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.2)
    • Android SDK at /usr/local/share/android-sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.2
    • ANDROID_HOME = /usr/local/share/android-sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4.1, Build version 9F2000
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 27.1.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] IntelliJ IDEA Ultimate Edition (version 2018.2.2)
    • IntelliJ at /Users/noel/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 27.1.3
    • Dart plugin version 182.4129.13

[✓] Connected devices (1 available)
    • Android SDK built for x86 64 • emulator-5554 • android-x64 • Android 9 (API 28) (emulator)

• No issues found!
```
Though, these gifs are screen recordings of the vanilla messaging app from the Android emulator.

**Comment by zoechi** [comment]
> Should be this closed as a result of #21715?

@HansMuller ?

**Comment by HansMuller** [comment]
@zeochi: thanks for the reminder. Support for Android P transitions was added in #21715.

I'm going to close this issue.
